### PR TITLE
Publish docs for the proc_macro crate

### LINKED ISSUE #38749: Publish docs for `proc_macro` crate
Documentation for the `proc_macro` crate currently does not appear in nightly builds, at least for Windows. This isn't too big of a deal for now as the API is quite simple, but it would still be nice if this crate appeared in the documentation.

### PR DESCRIPTION
**Pull request by SimonSapin**
Fix #38749

### CODE REVIEW

File: src/bootstrap/doc.rs [hunk 1]
```diff
@@ -162,7 +162,7 @@ pub fn std(build: &Build, stage: u32, target: &str) {
         cargo.arg("-p").arg("std");
     } else {
         cargo.arg("--no-deps");
-        for krate in &["alloc", "collections", "core", "std", "std_unicode"] {
+        for krate in &["alloc", "collections", "core", "std", "std_unicode", "proc_macro"] {
```
Comment (abonander) [inline src/bootstrap/doc.rs]: Shouldn't this be a const near the top of the file instead of a magic number / magic slice?

**Comment by est31** [comment]
I've tested it, it works!

**Review by est31 — APPROVED** [review]

File: src/bootstrap/doc.rs (hunk 1 above)
Comment (SimonSapin) [inline src/bootstrap/doc.rs]: Done.

**Comment by SimonSapin** [comment]
I made this PR as a drive-by on the assumption it was a one-line fix, but it sounds like it’ll require more time that I’m willing to spend on this.

So anyone feel free to take this over.

**Comment by alexcrichton** [comment]
Ok, in that case I'll close this for now (to help clear out the queue), but we can always reopen if necessary

**Pull request closed without merging**